client/src/components/SignUpAccountPage.jsx

```diff
@@ -0,0 +1,149 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { accountPageSubmit, createUser } from '../actions/actions.js';
+
+export class SignUpAccountPage extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      firstName: null,
+      lastName: null,
+      phone: null,
+      password: null,
+      passwordVerify: null,
+      fieldErrors: false
+    };
+
+    this.onClickSubmit = this.onClickSubmit.bind(this);
+  }
+
+  onClickSubmit() {
+    // this.props.dispatch(accountPageSubmit());
+    // console.log(this.state);
+
+    if (this.state.firstName === null ||
+      this.state.lastName === null ||
+      this.state.phone === null ||
+      this.state.password === null ||
+      this.state.password !== this.state.passwordVerify) {
+      this.setState({
+        fieldErrors: true
+      })
```

Comment:
This function is another place where destructuring could make things a lot cleaner.